test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp

```diff
@@ -0,0 +1,239 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alignment/matrix/debug_matrix.hpp>
+#include <seqan3/alignment/matrix/edit_distance_score_matrix_full.hpp>
+#include <seqan3/alignment/matrix/matrix_concept.hpp>
+#include <seqan3/test/pretty_printing.hpp>
+
+#include "edit_distance_score_matrix.hpp"
+
+TEST(global_max_errors, empty)
+{
+    matrix_type<false, true> matrix{1u};
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{}};
+
+    EXPECT_EQ(result, expect);
+}
+
+TEST(global_max_errors, epsilon)
+{
+    matrix_type<false, true> matrix{1u};
+
+    matrix.add_column({}, {}, 1u);
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{-0}};
+
+    EXPECT_EQ(result, expect);
+}
+
+TEST(global_max_errors, epsilon_row)
+{
+    matrix_type<false, true> matrix{1u};
+
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 0u);
+    matrix.add_column({}, {}, 0u);
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{-0, -1, -2,INF,INF}};
+
+    EXPECT_EQ(result, expect);
+}
+
+TEST(global_max_errors, single_word_1)
```

Comment:
single_word_1?
The number has a meaning, or it is just an enumeration of different test cases?

Reply:
I fixed that in the follwup.